trace.go

```diff
@@ -374,6 +425,43 @@ func (e *TraceErr) GoString() string {
 	return e.DebugReport()
 }
 
+// CompactUserMessage formats the wraoped error message as a compact go-style
+// wrapped error message. This format is more user-friendly and contains no
+// new line nor tab (except the ones already present in the wrapped error).
+// For example:
+//
+//	trace.Wrap(trace.Wrap(trace.Errorf("foo"), "bar"), "baz")
+//
+// Will produce the following compact error:
+//
+//	foo: bar: baz
```

Comment:
imo we should return `baz: bar: foo` in this example, going from the outermost message to the innermost

if i have
```go
return trace.Wrap(login(), "logging in")
```
i would want this to be formatted like `logging in: incorrect username or password`, not the other way

i'm pretty sure this is actually what the code does, based on the tests, and the comment is just wrong